engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -2320,23 +2357,222 @@ static void LogFrameBufferError(GLenum status)
+
+        for (int i = 0; i < context.m_Paths.Size(); ++i)
+        {
+            // These should be null-terminated
+            char* namespace_path = context.Namespace(i);
+            uint32_t namespace_path_len = strlen(namespace_path);
+
+            if (namespace_path && strcmp(namespace_path, canonical_name_buffer) == 0)
+            {
+                // Matching namespace found
+                char* canonical_name = context.CanonicalName(i);
+                assert(canonical_name);
+
+                char* str_without_namespace = str + namespace_end + 1;
+
+                // Found a match, we now write either the name without the namespace,
+                // or the instance + name without namespace
+                if (strcmp(canonical_name, str_without_namespace) == 0)
+                {
+                    char* instance_name = context.InstanceName(i);
+                    char* write_ptr = canonical_name_buffer;
+
+                    if (instance_name && instance_name[0] != 0)
+                    {
+                        uint32_t instance_name_len = strlen(instance_name);
+                        memcpy(write_ptr, instance_name, instance_name_len);
+                        write_ptr[instance_name_len] = '.';
+                        write_ptr += instance_name_len + 1;
+                    }
+
+                    uint32_t canonical_name_len = strlen(canonical_name);
+                    memcpy(write_ptr, canonical_name, canonical_name_len);
+                    write_ptr[canonical_name_len] = 0;
+                    return canonical_name_buffer;
+                }
+            }
+        }
+
+        return str;
+    }
+
+    static void OpenGLBuildUniforms(OpenGLContext* context, OpenGLProgram* program, OpenGLShader** shaders, uint32_t num_shaders)
     {
         if (context->m_IsGles3Version)
         {
             BuildUniformBuffers((HContext)context, program, shaders, num_shaders);
         }
 
-        uint32_t texture_unit = 0;
         char uniform_name_buffer[256];
+        char canonical_name_buffer[256];
 
         GLint num_uniforms;
         GLuint program_handle = GetGLHandle((HContext)context, program->m_Id);
         glGetProgramiv(program_handle, GL_ACTIVE_UNIFORMS, &num_uniforms);
         CHECK_GL_ERROR;
 
-        program->m_Uniforms.SetCapacity(num_uniforms);
-        program->m_Uniforms.SetSize(num_uniforms);
+        // The OpenGL reflect will not return the instance name of a uniform block,
+        // so we need to look at the reflection data to get the actual canonical path for a uniform.
+        // I.e, if a uniform is specified as:
+        //
+        // uniform fs_uniforms {
+        //     PBRMaterial material;
+        // } my_material;
+        //
+        // OpenGL will yield the following binding:
+        // fs_uniforms.material
+        //
+        // But what we actually want is:
+        // my_material.material
+
+        const bool prepend_instance_name = false;
```

Comment:
For all other adapters we can add the instance name to the canonical path, but for OpenGL we can't..